AmplifyPlugins/DataStore/Tests/DataStoreHostApp/AWSDataStorePluginLazyLoadTests/LL12/HasOneParentChild/AWSDataStoreLazyLoadHasOneTests.swift

```diff
@@ -77,37 +77,120 @@ class AWSDataStoreLazyLoadHasOneTests: AWSDataStoreLazyLoadBaseTest {
+            for try await mutationEvent in mutationEvents {
+                if let version = mutationEvent.version,
+                   version == 1,
+                   let receivedChild = try? mutationEvent.decodeModel(as: HasOneChild.self),
+                   receivedChild.identifier == child.identifier
+                {
+                    await mutationEventReceived.fulfill()
+                }
+            }
+        }
+
+        let createRequest = GraphQLRequest<MutationSyncResult>.createMutation(of: child, modelSchema: HasOneChild.schema)
+        do {
+            _ = try await Amplify.API.mutate(request: createRequest)
+        } catch {
+            XCTFail("Failed to send mutation request \(error)")
         }
 
+        await waitForExpectations([mutationEventReceived], timeout: 60)
+        mutationEvents.cancel()
+    }
+
+    func testDeleteHasOneChild() async throws {
+        await setup(withModels: HasOneModels())
+        let child = HasOneChild()
+        let savedChild = try await saveAndWaitForSync(child)
+        try await assertModelExists(savedChild)
+
+        try await deleteAndWaitForSync(savedChild)
+        try await assertModelDoesNotExist(savedChild)
+    }
+
+    func testDeleteHasOneParent() async throws {
+        await setup(withModels: HasOneModels())
+        let parent = HasOneParent()
+        let savedParent = try await saveAndWaitForSync(parent)
+        try await assertModelExists(savedParent)
+
+        try await deleteAndWaitForSync(savedParent)
+        try await assertModelDoesNotExist(savedParent)
+    }
+
+    func testDeleteHasOneParentWithChild() async throws {
+        await setup(withModels: HasOneModels())
+        let child = HasOneChild()
+        let savedChild = try await saveAndWaitForSync(child)
+        let parent = HasOneParent(child: savedChild, hasOneParentChildId: savedChild.id)
+        let savedParent = try await saveAndWaitForSync(parent)
+
+        let queriedParent = try await query(for: savedParent)
+        XCTAssertEqual(queriedParent.hasOneParentChildId, savedChild.id)
+
+        try await deleteAndWaitForSync(savedParent)
+        try await assertModelDoesNotExist(savedParent)
+        try await assertModelExists(savedChild)
+    }
+
+    func testDeleteHasOneChildWithParent() async throws {
+        await setup(withModels: HasOneModels())
+        let child = HasOneChild()
+        let savedChild = try await saveAndWaitForSync(child)
+        let parent = HasOneParent(child: savedChild, hasOneParentChildId: savedChild.id)
+        let savedParent = try await saveAndWaitForSync(parent)
+
+        let queriedParent = try await query(for: savedParent)
+        XCTAssertEqual(queriedParent.hasOneParentChildId, savedChild.id)
+
+        try await deleteAndWaitForSync(savedChild)
+        try await assertModelDoesNotExist(savedChild)
+        try await assertModelExists(savedParent)
+
+        let childIdInParent = try await query(for: savedParent).child?.id
```

Comment:
nit: do we need to fetch id here?

Reply:
oh sorry, i meant we can do something like this. i thought the whole child object will be `nil`
```
let childIdInParent = try await query(for: savedParent).child?
XCTAssertEqual(childIdInParent, nil)
```